### PR TITLE
Update strava.js

### LINKED ISSUE #3314: Strava provider returns 400 bad request
### Question 💬

Hello,

I have created a vanilla Next.js project and added version 4 of next-auth. I am trying to use the Strava provider, but I am receiving a 400 bad request. I am not sure if this is a bug, or a mistake I have made, any help would be appreciated. I have done the following. 

- Created an API/Oauth application on Strava.

strava_01
strava_02

- Populated __app.js_ with the following.
```
import '../styles/globals.css'

import { SessionProvider } from "next-auth/react"

export default function App({
  Component, pageProps: { session, ...pageProps }
}) {
  return (
    <SessionProvider session={session}>
      <Component {...pageProps}/>
    </SessionProvider>
  )
}
```
- Populated _index.js_ with the following.
```
import styles from "../styles/Home.module.css";
import { useSession, signIn, signOut } from "next-auth/react"

export default function Component() {
  const { data: session } = useSession()
  if(session) {
    return <>
      Signed in as {session.user.email} <br/>
      <button onClick={() => signOut()}>Sign out</button>
    </>
  }
  return <>
    Not signed in <br/>
    <button onClick={() => signIn()}>Sign in</button>
  </>
}
```

- Created _./api/auth/[...nextauth].js_ and populated it with the following.
_Ref: [https://next-auth.js.org/providers/strava](https://next-auth.js.org/providers/strava)_

```
import NextAuth from 'next-auth'
import StravaProvider from 'next-auth/providers/strava'

export default NextAuth({
  providers: [
    StravaProvider({
      clientId: process.env.STRAVA_CLIENT_ID,
      clientSecret: process.env.STRAVA_CLIENT_SECRET,
    })
  ]
})
```

- Created _./.env.local_ file and populated it with the following. I have also tried passing in the environment variables as string quoted.
```
STRAVA_CLIENT_ID=12345
STRAVA_CLIENT_SECRET=examplesecret
NEXTAUTH_URL=http://localhost:3000/
```

When I try logging in using Strava with either the user I created the application under, or a separate Strava user I created for testing, I receive the following error.

```
[next-auth][error][OAUTH_CALLBACK_ERROR] 
https://next-auth.js.org/errors#oauth_callback_error expected 200 OK, got: 400 Bad Request {
  error: {
    message: 'expected 200 OK, got: 400 Bad Request',
    stack: 'OPError: expected 200 OK, got: 400 Bad Request\n' +
      '    at processResponse (/scratch/fitness-dashboard/fitness-dashboard/node_modules/openid-client/lib/helpers/process_response.js:48:11)\n' +
      '    at Client.grant (/scratch/fitness-dashboard/fitness-dashboard/node_modules/openid-client/lib/client.js:1265:26)\n' +
      '    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n' +
      '    at async Client.oauthCallback (/scratch/fitness-dashboard/fitness-dashboard/node_modules/openid-client/lib/client.js:561:24)\n' +
      '    at async oAuthCallback (/scratch/fitness-dashboard/fitness-dashboard/node_modules/next-auth/core/lib/oauth/callback.js:114:16)\n' +
      '    at async Object.callback (/scratch/fitness-dashboard/fitness-dashboard/node_modules/next-auth/core/routes/callback.js:50:11)\n' +
      '    at async NextAuthHandler (/scratch/fitness-dashboard/fitness-dashboard/node_modules/next-auth/core/index.js:132:28)\n' +
      '    at async NextAuthNextHandler (/scratch/fitness-dashboard/fitness-dashboard/node_modules/next-auth/next/index.js:16:19)\n' +
      '    at async /scratch/fitness-dashboard/fitness-dashboard/node_modules/next-auth/next/index.js:52:32\n' +
      '    at async Object.apiResolver (/scratch/fitness-dashboard/fitness-dashboard/node_modules/next/dist/server/api-utils.js:102:9)',
    name: 'OPError'
  },
  providerId: 'strava',
  message: 'expected 200 OK, got: 400 Bad Request'
}
[next-auth][error][CALLBACK_OAUTH_ERROR] 
https://next-auth.js.org/errors#callback_oauth_error expected 200 OK, got: 400 Bad Request OPError: expected 200 OK, got: 400 Bad Request
    at processResponse (/scratch/fitness-dashboard/fitness-dashboard/node_modules/openid-client/lib/helpers/process_response.js:48:11)
    at Client.grant (/scratch/fitness-dashboard/fitness-dashboard/node_modules/openid-client/lib/client.js:1265:26)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Client.oauthCallback (/scratch/fitness-dashboard/fitness-dashboard/node_modules/openid-client/lib/client.js:561:24)
    at async oAuthCallback (/scratch/fitness-dashboard/fitness-dashboard/node_modules/next-auth/core/lib/oauth/callback.js:114:16)
    at async Object.callback (/scratch/fitness-dashboard/fitness-dashboard/node_modules/next-auth/core/routes/callback.js:50:11)
    at async NextAuthHandler (/scratch/fitness-dashboard/fitness-dashboard/node_modules/next-auth/core/index.js:132:28)
    at async NextAuthNextHandler (/scratch/fitness-dashboard/fitness-dashboard/node_modules/next-auth/next/index.js:16:19)
    at async /scratch/fitness-dashboard/fitness-dashboard/node_modules/next-auth/next/index.js:52:32
    at async Object.apiResolver (/scratch/fitness-dashboard/fitness-dashboard/node_modules/next/dist/server/api-utils.js:102:9) {
  name: 'OAuthCallbackError',
  code: undefined
}
[next-auth][warn][NO_SECRET] 
```

### PR DESCRIPTION
**Pull request by krautwigundrueben**
Auth with Strava was throwing errors before, this works.

did not work at all before, now works

## Checklist 🧢

- [ ] Documentation
- [x] Tests
- [x] Ready to be merged

yes, is tested successfully and can be merged

### CODE REVIEW

File: src/providers/strava.js [hunk 1]
```diff
@@ -4,17 +4,54 @@ export default function Strava(options) {
...
+        const { refresh_token, access_token, token_type, expires_at } = await response.json();
+
+        return {
+          tokens: {
+            refresh_token,
+            access_token,
+            token_type, 
+            expires_at,
+          },
+        };
```
Comment (balazsorban44) [inline src/providers/strava.js:41]: This part is most certainly unnecessary. If the configuration for Strava does not match the defaults, maybe the `client` provider config option might be of use.

https://next-auth.js.org/configuration/providers/oauth#client-option

File: src/providers/strava.js [hunk 2]
```diff
@@ -4,17 +4,54 @@ export default function Strava(options) {
...
+      },
+    },
     userinfo: "https://www.strava.com/api/v3/athlete",
+
```
Comment (balazsorban44) [inline src/providers/strava.js:45]:
```suggestion
```

File: src/providers/strava.js [hunk 3]
```diff
@@ -4,17 +4,54 @@ export default function Strava(options) {
...
     profile(profile) {
       return {
         id: profile.id,
         name: profile.firstname,
         email: null,
-        image: profile.profile,
-      }
+        image: profile.profile
+      };
```
Comment (balazsorban44) [inline src/providers/strava.js:52]:
```suggestion
        image: profile.profile,
      }
```

File: src/providers/strava.js [hunk 4]
```diff
@@ -4,17 +4,54 @@ export default function Strava(options) {
...
     },
-    options,
-  }
+
+    options
+  };
```
Comment (balazsorban44) [inline src/providers/strava.js:56]:
```suggestion
    options,
  }
```

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Thanks! Could you please address my comments?

To avoid linting mismatches, I suggest you run Prettier before committing.

Furthermore, I would like to ask if you would be interested in converting this provider to TypeScript? See this folder for several examples: https://github.com/nextauthjs/next-auth/tree/main/src/providers

I would also like to ask you to comment some kind of proof that the setup is actually doing its job. You can look merged PRs as an example from the table in the top description at #2524 

This would allow us to gain confidence that your PR actually works and that we can refer to it later on until we find a better way to automatically test provider integrations.

File: src/providers/strava.js (hunk 1 above)
Comment (gempi) [reply src/providers/strava.js:41]: @krautwigundrueben you can just add:

```js
client: {
  token_endpoint_auth_method: "client_secret_post",
}
```

**Review by krautwigundrueben — COMMENTED** [review]
I changed a couple of things according to your input - thanks!

**Comment by krautwigundrueben** [comment]
@balazsorban44 
> Furthermore, I would like to ask if you would be interested in converting this provider to TypeScript? See this folder for several examples: https://github.com/nextauthjs/next-auth/tree/main/src/providers
Sorry, but I don't really feel entitled to convert this right now as I've never done anything in TS. If you have enough time with this, I might get to it within weeks, but you'd rather ask someone else, please.

> I would also like to ask you to comment some kind of proof that the setup is actually doing its job. You can look merged PRs as an example from the table in the top description at #2524
proof for successful testing see this:
testing-proof-strava

**Comment by balazsorban44** [comment]
Thanks, I appreciate it. Let's get this merged for now then!

**Review by balazsorban44 — APPROVED** [review]

**Pull request merged**